presto-hive/src/main/java/com/facebook/presto/hive/metastore/ThriftHiveMetastore.java

```diff
@@ -582,6 +591,16 @@ public void grantTablePrivileges(String databaseName, String tableName, String g
     public void revokeTablePrivileges(String databaseName, String tableName, String grantee, Set<PrivilegeGrantInfo> privilegeGrantInfoSet)
     {
         try {
+            Set<HivePrivilegeInfo> existingPrivileges = getTablePrivileges(grantee, databaseName, tableName);
+
+            Set<PrivilegeGrantInfo> privilegesToRevoke = privilegeGrantInfoSet.stream()
+                    .filter(requestedPrivilege -> existingPrivileges.containsAll(parsePrivilege(requestedPrivilege)))
```

Comment:
Same comment here about renaming `requestedPrivilege` to just `privilege`.

Also isn't the filter condition supposed to be a negation so that you're revoking privilege that DON'T already exist?